documentation/examples/remote_storage/remote_storage_adapter/main.go

```diff
@@ -217,20 +217,8 @@ func serve(addr string, writers []writer, readers []reader) error {
 	})
 
 	http.HandleFunc("/read", func(w http.ResponseWriter, r *http.Request) {
-		compressed, err := ioutil.ReadAll(r.Body)
+		req, err := remote.DecodeRequest(r)
```

Comment:
The point of moving the protos was so remote adapters didn't have a dependency on Prometheus internals. If we're going to offer utility functions they need to be de-coupled.